enhancements/network/enable-flows-collection.md

```diff
@@ -0,0 +1,362 @@
+  time.  If  not  specified,  defaults to 0. If 0, caching is disabled.
+
+* `sampling`: optional integer, in range 1 to 4,294,967,295
+  The  rate  at  which  packets should be sampled and sent to each
+  target collector. If not specified, defaults to 400, which means
+  one  out of 400 packets, on average, will be sent to each target
+  collector.
+
+Other arguments might be considered, such as `enable-input-sampling`, `enable-output-sampling`,
+or `enable-tunnel-sampling`.
+
+The configuration should be also exposed in the Console UI, along with the current
+emission status (see previous subsection).
+
+### Implementation Details/Notes/Constraints [optional]
+
+In order to ensure an optimal usage of the network, we should restrict flow's
+emission-collection traffic to the local node.
+
+According to the [architectural description of the flows' emitter/collectors pipeline](
+https://docs.google.com/document/d/1kzNfTLXhMolu8VTcH0m9hmvFluN9yKMy7TSNIXJVTrA/edit#heading=h.kulf936ct32p),
+the collector will be deployed as a DaemonSet, to ensure that there will be an
+instance of the collector for each node.
+
+Exposing the collector endpoint as a Kubernetes ClusterIP Service would cause the traffic
+to be distributed across all the nodes (there is a `ServiceInternalTrafficPolicy`
+cluster-wide feature gate but, at the moment of writing this document, it is not implemented in
+OVN-Kubernetes, and it's beta in Kubernetes 1.22).
+
+To limit the flows' traffic to internal nodes, each collector should expose a node
+port, so each OVN instance will discover the Node IP (instead of the Service IP) and submit
+data to the NodeIP:NodePort address. The Node IP can be discovered.
+
+From the user-side configuration of the IPFix cache and sampling, we should use a `ConfigMap` and
+expose it to the CNO with read permissions. The CNO would configure OVN-Kubernetes with the configmap
+data.
+
+Given the aforementioned details, following subsections depicts the work to do,
+grouped in different tasks.
+
+#### Task 1: setup a new configurable CRD that is readable by the CNO
+
+[Current `Network` operator configuration allows setting a static host:port address](
+https://docs.okd.io/latest/networking/ovn_kubernetes_network_provider/tracking-network-flows.html#nw-network-flows-object_tracking-network-flows).
+This information is forwarded to each OVN instance, on each cluster node. For example:
+
+```yaml
+apiVersion: operator.openshift.io/v1
+kind: Network
+metadata:
+  name: cluster
+spec:
+  exportNetworkFlows:
+    collectors:
+      - 192.168.1.99:2056
+```
+
+However, we need to set this address dynamically to ensure that each pod will forward
+the flows to the collector in its same node.
+
+While the above `Network` configuration is still valid for the current use cases, the Network
+Observability components will provide a new CRD whose configuration is exposed to the CNO, which
+will use it to configure the OVN-Kubernetes operator. For example:
```

Comment:
I've drawn a diagram that would summarize the interaction between the CNO and our new NOO (network observability operator) : https://docs.google.com/drawings/d/1zTbGOHUZlm3mKUcnb3s7SdXQbJcyoOIGieDzYOjxlxg/edit

Basically, here's how it would work:

NOO creates a ConfigMap for OvS, and CNO is able to understand it and will be watching it for changes.

For the CNO, this config map content sums up with any OvS configuration that potentially exists in the Network CR. For example, if a user has configured an IP for netflow exports in the Network CR AND has configured a Netobs CR for the NOO, OvS will export flows to both destinations.

Currently, the CNO interacts with the OVN pods via the DaemonSet definition and environment variables set on some of the OVN containers. New environment variables would be introduced to account for the new configuration needed by the NOO.

The OVN pods need to use this new config while setting up the OvS flow exports.